tla-assignments/src/main/scala/at/forsyte/apalache/tla/assignments/SymbTransGenerator.scala

```diff
@@ -272,10 +291,15 @@ class SymbTransGenerator(tracker: TransformationTracker) {
     def getTransitions(
         ex: TlaEx,
         strategy: StrategyType,
-        selections: SelMapType): Seq[SymbTrans] =
-      selections(ex.ID).map { s =>
+        selections: SortedSelMapType): Seq[SymbTrans] = {
+      // It is crucial to have selections sorted here, to have a deterministic output.
+      // Otherwise, `sliceWith` produces UIDs in different orders on different runs.
+      // As a result, multiple runs of the same spec would produce different orderings of transitions,
+      // which is highly undesirable.
+      selections(ex.ID).toSeq.map { s =>
```

Comment:
Since selections(ex.ID) is a `SortedSet`, cannot we

```suggestion
      selections(ex.ID).map { s =>
```

Reply:
The resulting type is `Seq[...]`, so we have to convert it.